plugins/techdocs/src/reader/components/Reader.tsx

```diff
@@ -47,12 +47,32 @@ export const Reader = ({ entityId, onReady }: Props) => {
 
   const techdocsStorageApi = useApi(techdocsStorageApiRef);
   const [shadowDomRef, shadowRoot] = useShadowDom();
+  const [sidebars, setSidebars] = useState<HTMLElement[]>();
   const navigate = useNavigate();
 
   const { value, loading, error } = useAsync(async () => {
     return techdocsStorageApi.getEntityDocs({ kind, namespace, name }, path);
   }, [techdocsStorageApi, kind, namespace, name, path]);
 
+  useEffect(() => {
+    const updateSidebarPosition = () => {
+      if (!!shadowRoot && !!shadowDomRef.current && !!sidebars) {
+        sidebars!.forEach(sidebar => {
+          const newTop = Math.max(
+            shadowDomRef.current!.getBoundingClientRect().top,
+            0,
+          );
+          sidebar.style.top = `${newTop}px`;
+        });
+      }
+    };
+    updateSidebarPosition();
+    document.addEventListener('scroll', updateSidebarPosition);
```

Comment:
Just noting: I was concerned that this would be problematic from a performance perspective, but I didn't see anything concerning in chrome dev tools performance pane, I'm guessing because setting position on a fixed element doesn't cause reflow?  🤷‍♂️ 

Not that there's much we can do about this anyway, I think.  Given the shadow DOM implementation, this is kind of our only option, eh?

Reply:
Yes,  fixed elements are outside the flow so don't cause reflows.